pkg/controller/build/build_controller.go

```diff
@@ -0,0 +1,1297 @@
+	}
+
+	updatedPool.Status = pool.Status
+
+	return statusFunc(updatedPool)
+}
+
+// Machine Config Pools
+
+func (ctrl *Controller) addMachineConfigPool(obj interface{}) {
+	pool := obj.(*mcfgv1.MachineConfigPool).DeepCopy()
+	klog.V(4).Infof("Adding MachineConfigPool %s", pool.Name)
+	ctrl.enqueueMachineConfigPool(pool)
+}
+
+// Prepares all of the objects needed to perform an image build.
+func (ctrl *Controller) prepareMachineConfigForPool(ibr ImageBuildRequest) error {
+	mc, err := ctrl.mcfgclient.MachineconfigurationV1().MachineConfigs().Get(context.TODO(), ibr.Pool.Spec.Configuration.Name, metav1.GetOptions{})
+	if err != nil {
+		return fmt.Errorf("could not get MachineConfig %s: %w", ibr.Pool.Spec.Configuration.Name, err)
+	}
+
+	mcConfigMap, err := ibr.toConfigMap(mc)
+	if err != nil {
+		return fmt.Errorf("could not convert MachineConfig %s into ConfigMap: %w", mc.Name, err)
+	}
+
+	_, err = ctrl.kubeclient.CoreV1().ConfigMaps(ctrlcommon.MCONamespace).Create(context.TODO(), mcConfigMap, metav1.CreateOptions{})
+	if err != nil {
+		return fmt.Errorf("could not load rendered MachineConfig %s into configmap: %w", mcConfigMap.Name, err)
+	}
+
+	klog.Infof("Stored MachineConfig %s in ConfigMap %s for build", mc.Name, mcConfigMap.Name)
+
+	dockerfileConfigMap, err := ibr.dockerfileToConfigMap()
+	if err != nil {
+		return fmt.Errorf("could not generate Dockerfile ConfigMap: %w", err)
+	}
+
+	_, err = ctrl.kubeclient.CoreV1().ConfigMaps(ctrlcommon.MCONamespace).Create(context.TODO(), dockerfileConfigMap, metav1.CreateOptions{})
+	if err != nil {
+		return fmt.Errorf("could not load rendered Dockerfile %s into configmap: %w", dockerfileConfigMap.Name, err)
+	}
+
+	klog.Infof("Stored Dockerfile for build %s in ConfigMap %s for build", ibr.getBuildName(), dockerfileConfigMap.Name)
+
+	return nil
+}
+
+// Determines if we should run a build, then starts a build pod to perform the
+// build, and updates the MachineConfigPool with an object reference for the
+// build pod.
+func (ctrl *Controller) startBuildForMachineConfigPool(pool *mcfgv1.MachineConfigPool) error {
+	osImageURLConfigMap, err := ctrl.kubeclient.CoreV1().ConfigMaps(ctrlcommon.MCONamespace).Get(context.TODO(), machineConfigOSImageURLConfigMapName, metav1.GetOptions{})
+	if err != nil {
+		return fmt.Errorf("could not get OS image URL: %w", err)
+	}
+
+	onClusterBuildConfigMap, err := ctrl.getOnClusterBuildConfig(pool)
+	if err != nil {
+		return fmt.Errorf("could not get configmap %q: %w", onClusterBuildConfigMapName, err)
+	}
+
+	ibr := newImageBuildRequestWithConfigMap(pool, osImageURLConfigMap, onClusterBuildConfigMap)
+
+	return ctrl.handleImageBuildRequest(ibr)
+}
+
+// Gets the ConfigMap which specifies the name of the base image pull secret, final image pull secret, and final image pullspec.
+func (ctrl *Controller) getOnClusterBuildConfig(pool *mcfgv1.MachineConfigPool) (*corev1.ConfigMap, error) {
```

Comment:
I didn't think to wire up a cleanup mechanism to it because it's shared across all of the builds and I didn't want something to delete it while something else was in progress. Although arguably, something that the BuildController should do is determine if there are no builds in progress and clean up after itself. But I suppose that I could do that in a followup PR as well as wire up cleanup functions to the other objects as well.